src/demo/data.js

```diff
@@ -2,51 +2,153 @@ import faker from 'faker/locale/en';
+  const totalCurrentEnrollements = enrollments.filter(currentEnrollment => currentEnrollment.user_email === enrollment.user_email).length; // eslint-disable-line max-len
+  const totalCompletedEnrollements = enrollments.filter(currentEnrollment => currentEnrollment.has_passed === true && currentEnrollment.user_email === enrollment.user_email).length; // eslint-disable-line max-len
   return {
-    id: idx,
-    user_email: faker.internet.email(firstName, lastName, provider),
-    course_title: course.title,
-    course_price: course.price,
-    course_start: courseStart,
-    course_end: courseEnd,
-    passed_timestamp: Math.random() <= hasPassed ? faker.date.between(courseStart, courseEnd)
-      : null,
-    has_passed: hasPassed,
-    current_grade: grade / 100,
-    last_activity_date: faker.date.recent(Math.random() <= percentActive ? 7 : 45),
+    user_email: enrollment.user_email,
+    enrollment_count: totalCurrentEnrollements,
+    course_completion_count: totalCompletedEnrollements,
+    last_activity_date: enrollment.last_activity_date,
   };
 });
 
+const filterUnenrolledRegisteredLearners = () => enrollments.map(enrollment => ({
+  user_email: enrollment.user_email,
+  user_account_creation_timestamp: faker.date.past(),
+}));
+
+const filterLearnerActivity = (learnerEnrollments, filterOption) => {
+  let filteredLearnerEnrollments = learnerEnrollments;
+  const lastActivityDate = new Date();
+  if (filterOption === 'active_past_week') {
+    lastActivityDate.setDate(lastActivityDate.getDate() - 7);
+    filteredLearnerEnrollments = learnerEnrollments.filter(enrollment => enrollment.last_activity_date.getTime() >= lastActivityDate.getTime()); // eslint-disable-line max-len
+  } else if (filterOption === 'inactive_past_week') {
+    lastActivityDate.setDate(lastActivityDate.getDate() - 7);
+    filteredLearnerEnrollments = learnerEnrollments.filter(enrollment => enrollment.last_activity_date.getTime() <= lastActivityDate.getTime()); // eslint-disable-line max-len
+  } else if (filterOption === 'inactive_past_month') {
+    lastActivityDate.setDate(lastActivityDate.getDate() - 30);
+    filteredLearnerEnrollments = learnerEnrollments.filter(enrollment => enrollment.last_activity_date.getTime() <= lastActivityDate.getTime()); // eslint-disable-line max-len
+  }
+  return filteredLearnerEnrollments;
+};
+
+const filterPastWeekCompletions = (pastWeekEnrollments) => {
+  const today = new Date();
+  const passedDateTime = new Date().setDate(today.getDate() - 7);
+  return pastWeekEnrollments.filter(enrollment => enrollment.has_passed &&
+    enrollment.passed_timestamp.getTime() >= passedDateTime &&
+    enrollment.passed_timestamp.getTime() <= today.getTime());
+};
+
+const filterCompletedLearnerCourses = () => enrollments.filter(enrollment => enrollment.has_passed).map((enrollment) => { // eslint-disable-line max-len
```

Comment:
Can this line be formatted to avoid using `// eslint-disable-line max-len`? It would probably also make it easier to read.